apps/src/lib/kits/maker/WebSerialPortWrapper.js

```diff
@@ -24,67 +24,66 @@ export default class WebSerialPortWrapper extends EventEmitter {
     // TODO - not sure if this is used in Maker Toolkit yet
   }
 
-  /** Opens the serial port from Micro:Bit and starts reading from port.
-   * We are opening the port here and not reading from or writing to the port at this point in
-   * implementation. Later on when the entire Micro:WebSerial pathway is implemented,
-   * this function may be removed logic added for the MB in the open() function below,
-   * if appropriate.
-   */
-  openMBPort() {
+  // Opens the serial port from Micro:Bit and starts reading from port.
+  async openMBPort() {
     if (this.portOpen) {
       throw new Error(`Requested port is already open.`);
     }
     return this.port
       .open({baudRate: SERIAL_BAUD})
-      .then(() => {
-        this.writer = this.port.writable.getWriter();
-        this.reader = this.port.readable.getReader();
-      })
       .then(() => {
         this.portOpen = true;
         this.emit('open');
+        this.writer = this.port.writable.getWriter();
+        this.reader = this.port.readable.getReader();
+
+        this.readLoop();
```

Comment:
Want to make sure I understand this - I'm guessing you want to block on `readLoop` until proceeding to the next call in the promise chain right? Or not necessarily? I _think_ as currently written, this will just proceed to the next call without waiting. However if you do want to wait, I think you'd just need to `return this.readLoop()`, and then the next promise chain call would only happen once readLoop has returned. Though I could definitely be misunderstanding something about how async functions work in promises!

Reply:
I don't want to block on `readLoop` here. This loop should run continuously until the port is closed.

Reply:
Gotcha, thanks for clarifying! In that case, mind leaving a comment here that we want to start this process but not block on it? Also maybe `startReadLoop`/`runReadLoop` might be clearer?